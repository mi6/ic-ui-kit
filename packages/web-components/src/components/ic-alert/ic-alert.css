@import "../../global/normalize.css";

:host {
  display: block;
}

.container {
  min-height: 3.5rem;
  border-radius: var(--ic-border-radius);
  position: relative;
  display: flex;
  align-items: center;
}

.container-neutral {
  background-color: var(--ic-status-unknown-light);
}

.container-info {
  background-color: var(--ic-status-info-light);
}

.container-warning {
  background-color: var(--ic-status-warning-light);
}

.container-error {
  background-color: var(--ic-status-error-light);
}

.container-success {
  background-color: var(--ic-status-success-light);
}

.alert-icon-container {
  height: 100%;
  display: flex;
  align-items: center;
}

.divider {
  height: 100%;
  width: var(--ic-space-xs);
  border-radius: var(--ic-space-xxxs) 0 0 var(--ic-space-xxxs);
  position: absolute;
}

.divider-neutral {
  background-color: var(--ic-status-unknown);
}

.divider-info {
  background-color: var(--ic-status-info);
}

.divider-warning {
  background-color: var(--ic-status-warning);
}

.divider-error {
  background-color: var(--ic-status-error);
}

.divider-success {
  background-color: var(--ic-status-success);
}

.alert-icon {
  height: var(--ic-space-lg);
  width: 1.375rem;
  margin-left: 1.125rem;
}

.alert-icon > svg {
  height: var(--ic-space-lg);
  width: var(--ic-space-lg);
  display: inline-block;
}

.icon-neutral > svg {
  fill: var(--ic-status-unknown);
}

:host([variant="info"]) .alert-icon svg {
  fill: var(--ic-status-info);
}

:host([variant="warning"]) .alert-icon svg {
  fill: var(--ic-status-warning);
}

:host([variant="error"]) .alert-icon svg {
  fill: var(--ic-status-error);
}

:host([variant="success"]) .alert-icon svg {
  fill: var(--ic-status-success);
}

.alert-content {
  display: flex;
  align-items: center;
  margin-left: 0.625rem;
  width: 100%;
}

.alert-message {
  display: flex;
  align-items: center;
  padding: var(--ic-space-xs) var(--ic-space-xs) var(--ic-space-xs) 0;
  flex: 1;
}

.alert-message-title-above {
  display: inline;
  font-size: 0;
}

.alert-title {
  margin-right: var(--ic-space-xs);
}

.alert-title-above {
  white-space: normal;
}

.alert-action-container {
  margin-right: var(--ic-space-xs);
  display: flex;
  align-items: center;
}

.dismiss-icon {
  margin-right: var(--ic-space-xxxs);
  margin-left: -0.375rem;
  padding: 0.375rem;
  border: none;
  border-radius: 50%;
  background-color: inherit;
  display: flex;
  align-items: center;
  justify-content: center;
}

.dismiss-icon:hover {
  cursor: pointer;
}

@media (max-width: 628px) {
  .alert-content {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }

  .alert-message {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }

  .alert-title {
    white-space: normal;
  }

  .alert-action-container {
    margin-bottom: var(--ic-space-xs);
  }
}

@media (forced-colors: active) {
  .container {
    border: var(--ic-hc-border);
  }
}
