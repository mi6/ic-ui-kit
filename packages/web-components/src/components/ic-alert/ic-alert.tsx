import {
  Component,
  Element,
  Event,
  EventEmitter,
  Host,
  State,
  Listen,
  Prop,
  h,
} from "@stencil/core";
import closeIcon from "../../assets/close-icon.svg";
import { isSlotUsed } from "../../utils/helpers";
import { IcThemeForegroundEnum, IcStatusVariants } from "../../utils/types";
import { VARIANT_ICONS } from "../../utils/constants";

/**
 * @slot message - Content is placed to the right of the title.
 * @slot action - Content is placed to the right of the message.
 */
@Component({
  tag: "ic-alert",
  styleUrl: "ic-alert.css",
  shadow: true,
})
export class Alert {
  @Element() el: HTMLIcAlertElement;

  @State() alertTitleWrap: boolean = false;
  @State() visible: boolean = true;

  /**
   * If `true`, the alert will have the 'alert' ARIA role and will be announced to screen readers.
   */
  @Prop() announced?: boolean = true;

  /**
   * If `true`, the alert will have a close icon at the end to dismiss it.
   */
  @Prop() dismissible?: boolean = false;

  /**
   * The optional title to display at the start of the alert.
   */
  @Prop() heading?: string = "";

  /**
   * The main body message of the alert.
   */
  @Prop() message?: string;

  /**
   * If `true`, the title and message will appear inline instead of above and below.
   */
  @Prop() titleAbove?: boolean = false;

  /**
   * The variant of the alert which will be rendered.
   */
  @Prop() variant?: IcStatusVariants = "neutral";

  /**
   * @deprecated This event should not be used anymore. Use icDismiss instead.
   */
  @Event() dismiss: EventEmitter<void>;

  /**
   * Is emitted when the user dismisses the alert.
   */
  @Event() icDismiss: EventEmitter<void>;

  componentDidLoad(): void {
    this.alertTitleShouldWrap();
  }

  @Listen("icDismiss", { capture: true })
  handleClick(): void {
    this.visible = !this.visible;
  }

  private dismissAction = (): void => {
    this.dismiss.emit();
    this.icDismiss.emit();
  };

  private alertTitleShouldWrap() {
    const titleHeight =
      this.el.shadowRoot.querySelector(".alert-title")?.clientHeight;
    if (titleHeight > 24) this.alertTitleWrap = true;
  }

  render() {
    const {
      variant,
      heading,
      message,
      titleAbove,
      dismissible,
      announced,
      visible,
    } = this;

    return (
      visible && (
        <Host
          role={announced && "alert"}
          class={{
            [IcThemeForegroundEnum.Dark]: true,
          }}
        >
          <div
            class={{
              ["container"]: true,
              [`container-${variant}`]: true,
            }}
          >
            <div class="alert-icon-container">
              <div
                class={{
                  ["divider"]: true,
                  [`divider-${variant}`]: true,
                }}
              ></div>
              <span
                class={{
                  ["alert-icon"]: true,
                  ["svg-container"]: true,
                  [`icon-${variant}`]: true,
                }}
                innerHTML={VARIANT_ICONS[variant].icon}
              ></span>
            </div>
            <div class="alert-content">
              <div
                class={{
                  ["alert-message"]: true,
                  ["alert-message-title-above"]:
                    titleAbove || this.alertTitleWrap,
                }}
              >
                {heading && (
                  <ic-typography
                    class={{
                      ["alert-title"]: true,
                      ["alert-title-above"]: titleAbove || this.alertTitleWrap,
                    }}
                    variant="subtitle-large"
                  >
                    <p>{heading}</p>
                  </ic-typography>
                )}
                <slot name="message">
                  <ic-typography variant="body">{message}</ic-typography>
                </slot>
              </div>
              {isSlotUsed(this.el, "action") && (
                <div class="alert-action-container">
                  <slot name="action" />
                </div>
              )}
            </div>
            <div class="dismiss-icon-container">
              {dismissible && (
                <ic-button
                  class={{
                    ["svg-container"]: true,
                    ["dismiss-icon"]: true,
                  }}
                  innerHTML={closeIcon}
                  onClick={this.dismissAction}
                  variant="icon"
                  appearance={IcThemeForegroundEnum.Dark}
                  title="dismiss"
                ></ic-button>
              )}
            </div>
          </div>
        </Host>
      )
    );
  }
}
