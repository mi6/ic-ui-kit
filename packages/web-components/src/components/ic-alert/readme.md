# ic-alert



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description                                                                                   | Type                                                       | Default     |
| ------------- | ------------- | --------------------------------------------------------------------------------------------- | ---------------------------------------------------------- | ----------- |
| `announced`   | `announced`   | If `true`, the alert will have the 'alert' ARIA role and will be announced to screen readers. | `boolean`                                                  | `true`      |
| `dismissible` | `dismissible` | If `true`, the alert will have a close icon at the end to dismiss it.                         | `boolean`                                                  | `false`     |
| `heading`     | `heading`     | The optional title to display at the start of the alert.                                      | `string`                                                   | `""`        |
| `message`     | `message`     | The main body message of the alert.                                                           | `string`                                                   | `undefined` |
| `titleAbove`  | `title-above` | If `true`, the title and message will appear inline instead of above and below.               | `boolean`                                                  | `false`     |
| `variant`     | `variant`     | The variant of the alert which will be rendered.                                              | `"error" \| "info" \| "neutral" \| "success" \| "warning"` | `"neutral"` |


## Events

| Event       | Description                                                                                                             | Type                |
| ----------- | ----------------------------------------------------------------------------------------------------------------------- | ------------------- |
| `dismiss`   | <span style="color:red">**[DEPRECATED]**</span> This event should not be used anymore. Use icDismiss instead.<br/><br/> | `CustomEvent<void>` |
| `icDismiss` | Is emitted when the user dismisses the alert.                                                                           | `CustomEvent<void>` |


## Slots

| Slot        | Description                                    |
| ----------- | ---------------------------------------------- |
| `"action"`  | Content is placed to the right of the message. |
| `"message"` | Content is placed to the right of the title.   |


## Dependencies

### Used by

 - [ic-dialog](../ic-dialog)

### Depends on

- [ic-typography](../ic-typography)
- [ic-button](../ic-button)

### Graph
```mermaid
graph TD;
  ic-alert --> ic-typography
  ic-alert --> ic-button
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  ic-dialog --> ic-alert
  style ic-alert fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


