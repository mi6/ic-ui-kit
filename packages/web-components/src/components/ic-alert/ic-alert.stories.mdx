import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Description,
} from "@storybook/addon-docs";
import { html } from "lit-html";
import readme from "./readme.md";
import Link from "../ic-link/readme.md";
import Button from "../ic-button/readme.md";

<Meta title="Web Components/Alert" component="ic-alert" />

<Description markdown={readme} />

export const defaultArgs = {
  dismissible: false,
  heading: "Heading",
  message: "Message",
  titleAbove: false,
  variant: "neutral",
};

### Variants

<Canvas>
  <Story name="Variants" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
        heading="Neutral"
        message="This alert is for displaying miscellaneous messages"
      ></ic-alert>
      <ic-alert
        variant="info"
        heading="Info"
        message="This alert is for displaying information"
      ></ic-alert>
      <ic-alert
        variant="error"
        heading="Error"
        message="This alert is for displaying errors"
      ></ic-alert>
      <ic-alert
        variant="warning"
        heading="Warning"
        message="This alert is for displaying warnings"
      ></ic-alert>
      <ic-alert
        variant="success"
        heading="Success"
        message="This alert is for displaying success messages. It has a very long message with lots of text which goes over multiple
        lines so the alert should expand as well to make sure it doesn't overflow"
      ></ic-alert>`}
  </Story>
</Canvas>

### Message only

<Canvas>
  <Story name="Message only" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
      message="This alert is for displaying miscellaneous messages"
    ></ic-alert>`}
  </Story>
</Canvas>

### Dismissible

<Canvas>
  <Story name="Dismissible" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
      heading="Neutral"
      message="This alert is for displaying miscellaneous messages"
      dismissible="true"
    ></ic-alert>`}
  </Story>
</Canvas>

### With action

<Canvas>
  <Story name="With action" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
      heading="Neutral"
      message="This alert is for displaying miscellaneous messages"
      ><ic-button slot="action" variant="secondary">Button</ic-button></ic-alert
    >`}
  </Story>
</Canvas>

### Dismissible with action

<Canvas>
  <Story name="Dismissible with action" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
      heading="Neutral"
      message="This alert is for displaying miscellaneous messages"
      dismissible="true"
      ><ic-button slot="action" variant="secondary">Button</ic-button></ic-alert
    >`}
  </Story>
</Canvas>

### Title above

<Canvas>
  <Story name="Title above" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
      heading="Neutral"
      message="This alert is for displaying miscellaneous messages"
      title-above="true"
    ></ic-alert>`}
  </Story>
</Canvas>

### Responsive title

<Canvas>
  <Story name="Responsive title" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert
        heading="This title is very long so should force the alert to add 'title-above'"
        message="This alert does not have 'title-above' added as a prop but it is forced to add it due to the length of the title"
      ></ic-alert>
      <ic-alert
        variant="success"
        heading="Shorter title"
        message="Neither of these has 'title-above' set to true"
      ></ic-alert>`}
  </Story>
</Canvas>

### Custom message

<Canvas>
  <Story name="Custom message" parameters={{ loki: { skip: true } }}>
    {html`<ic-alert heading="This alert uses a custom message slot">
      <ic-typography variant="body" slot="message">
        <p>
          This is some text and
          <ic-link href="/" inline>this is an inline link</ic-link> within the
          text.
        </p>
      </ic-typography>
    </ic-alert>`}
  </Story>
</Canvas>

### Custom message and title above

<Canvas>
  <Story
    name="Custom message and title above"
    parameters={{ loki: { skip: true } }}
  >
    {html`<ic-alert
      heading="Want to know more about our coffee?"
      title-above="true"
    >
      <span slot="message" style="font:var(--ic-font-body)">
        Go to our <ic-link href="/">coffee page</ic-link> to learn more.
      </ic-typography>
    </ic-alert>`}
  </Story>
</Canvas>

### Announced

<Canvas>
  <Story name="Announced" parameters={{ loki: { skip: true } }}>
    {html`
      <ic-alert
        id="alert"
        heading="Neutral"
        message="This alert is for displaying miscellaneous messages"
        announced="false"
      ></ic-alert>
      <br />
      <br />
      <ic-button variant="primary" size="small" onClick="onClick()" id="btn"
        >Toggle announced prop</ic-button
      >
      <script>
        const alert = document.getElementById("alert");
        const btn = document.getElementById("btn");
        btn.addEventListener("click", onClick);
        function onClick(e) {
          e.preventDefault();
          alert.announced
            ? (alert.announced = "false")
            : (alert.announced = "true");
        }
      </script>
    `}
  </Story>
</Canvas>

### Playground

<Canvas>
  <Story
    args={defaultArgs}
    argTypes={{
      variant: {
        options: ["neutral", "info", "warning", "error", "success"],
        control: { type: "select" },
      },
    }}
    name="Playground"
    parameters={{ loki: { skip: true } }}
  >
    {(args) =>
      html`<ic-alert
        dismissible=${args.dismissible}
        variant=${args.variant}
        heading=${args.heading}
        message=${args.message}
        title-above=${args.titleAbove}
      ></ic-alert>`
    }
  </Story>
</Canvas>
