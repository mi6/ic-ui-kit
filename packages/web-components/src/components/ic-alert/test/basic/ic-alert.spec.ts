import { newSpecPage } from "@stencil/core/testing";
import { Alert } from "../../ic-alert";

describe("ic-alert component", () => {
  it("should render with a heading when supplied", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert heading="Test heading"></ic-alert>`,
    });

    expect(page.root).toEqualHtml(`
        <ic-alert class="dark" heading="Test heading" role="alert">
            <mock:shadow-root>
                <div class="container container-neutral">
                    <div class="alert-icon-container">
                        <div class="divider divider-neutral"></div>
                        <span class="alert-icon icon-neutral svg-container">
                            svg
                        </span>
                    </div>
                    <div class="alert-content">
                        <div class="alert-message">
                            <ic-typography class="alert-title" variant="subtitle-large">
                                <p>
                                    Test heading
                                </p>
                            </ic-typography>
                            <slot name="message">
                                <ic-typography variant="body"></ic-typography>
                            </slot>
                        </div>
                    </div>
                    <div class="dismiss-icon-container">
                    </div>
                </div>
            </mock:shadow-root>
        </ic-alert>`);
  });

  it("should render with a message when supplied", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert message="Test message"></ic-alert>`,
    });

    expect(page.root).toEqualHtml(`
        <ic-alert class="dark" message="Test message" role="alert">
            <mock:shadow-root>
                <div class="container container-neutral">
                    <div class="alert-icon-container">
                    <div class="divider divider-neutral"></div>
                        <span class="alert-icon icon-neutral svg-container">
                            svg
                        </span>
                    </div>
                    <div class="alert-content">
                        <div class="alert-message">
                            <slot name="message">
                                <ic-typography variant="body">
                                    Test message
                                </ic-typography>
                            </slot>
                        </div>
                    </div>
                    <div class="dismiss-icon-container">
                    </div>
                </div>
            </mock:shadow-root>
        </ic-alert>`);
  });

  it("should render an action in the correct position when supplied", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert message="This has an action"><button slot="action"></button></ic-alert>`,
    });

    expect(page.root).toEqualHtml(`
        <ic-alert class="dark" message="This has an action" role="alert">
            <mock:shadow-root>
                <div class="container container-neutral">
                    <div class="alert-icon-container">
                        <div class="divider divider-neutral"></div>
                        <span class="alert-icon icon-neutral svg-container">
                            svg
                        </span>
                    </div>
                    <div class="alert-content">
                        <div class="alert-message">
                            <slot name="message">
                                <ic-typography variant="body">
                                    This has an action
                                </ic-typography>
                            </slot>
                        </div>
                        <div class="alert-action-container">
                            <slot name="action"></slot>
                        </div>
                    </div>
                    <div class="dismiss-icon-container">
                    </div>
                </div>
            </mock:shadow-root>
            <button slot="action"></button>
        </ic-alert>`);
  });

  it("should render with a dismiss icon when the prop is applied", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert message="This is dismissible" dismissible=true></ic-alert>`,
    });

    expect(page.root).toEqualHtml(`
            <ic-alert class="dark" message="This is dismissible" dismissible=true role="alert">
                <mock:shadow-root>
                    <div class="container container-neutral">
                        <div class="alert-icon-container">
                            <div class="divider divider-neutral"></div>
                            <span class="alert-icon icon-neutral svg-container">
                                svg
                            </span>
                        </div>
                        <div class="alert-content">
                            <div class="alert-message">
                                <slot name="message">
                                    <ic-typography variant="body">
                                        This is dismissible
                                    </ic-typography>
                                </slot>
                            </div>
                        </div>
                        <div class="dismiss-icon-container">
                            <ic-button appearance="dark" class="dismiss-icon svg-container" title="dismiss" variant="icon">
                                svg
                            </ic-button>
                        </div>
                    </div>
                </mock:shadow-root>
            </ic-alert>`);
  });

  it("should render an element in the message slot", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert heading="Using custom message"><ic-typography variant="h1" slot="message">Custom h1 message</ic-typography></ic-alert>`,
    });

    expect(page.root).toEqualHtml(`
        <ic-alert class="dark" heading="Using custom message" role="alert">
            <mock:shadow-root>
                <div class="container container-neutral">
                    <div class="alert-icon-container">
                        <div class="divider divider-neutral"></div>
                            <span class="alert-icon icon-neutral svg-container">
                                svg
                            </span>
                        </div>
                    <div class="alert-content">
                        <div class="alert-message">
                            <ic-typography class="alert-title" variant="subtitle-large">
                                <p>
                                    Using custom message
                                </p>
                            </ic-typography>
                            <slot name="message">
                                <ic-typography variant="body"></ic-typography>
                            </slot>
                        </div>
                    </div>
                    <div class="dismiss-icon-container"></div>
                </div>
            </mock:shadow-root>
            <ic-typography variant="h1" slot="message">Custom h1 message</ic-typography>
        </ic-alert>`);
  });

  it("should close on dismiss icon click", async () => {
    const page = await newSpecPage({
      components: [Alert],
      html: `<ic-alert message="This is dismissible" dismissible=true></ic-alert>`,
    });

    let alert = await page.root.shadowRoot.querySelector(
      "div.container-neutral"
    );
    const dismissButton = await page.root.shadowRoot.querySelector("ic-button");

    expect(alert).not.toBeNull();
    expect(dismissButton).not.toBeNull();

    await dismissButton.click();

    await page.waitForChanges();
    alert = await page.root.shadowRoot.querySelector("div.container-neutral");

    expect(alert).toBeNull();
  });
});
