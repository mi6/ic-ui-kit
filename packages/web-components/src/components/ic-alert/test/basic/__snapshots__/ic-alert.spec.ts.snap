// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-alert component should render an action in the correct position when supplied 1`] = `
<ic-alert class="ic-alert-variant-neutral" message="This has an action" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <slot name="message">
            <ic-typography variant="body">
              This has an action
            </ic-typography>
          </slot>
        </div>
        <div class="alert-action-container">
          <slot name="action"></slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
  <button slot="action"></button>
</ic-alert>
`;

exports[`ic-alert component should render an element in the message slot 1`] = `
<ic-alert class="ic-alert-variant-neutral" heading="Using custom message" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              Using custom message
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body"></ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
  <ic-typography slot="message" variant="h1">
    Custom h1 message
  </ic-typography>
</ic-alert>
`;

exports[`ic-alert component should render an icon in the neutral-icon slot 1`] = `
<ic-alert class="ic-alert-variant-neutral" heading="Using custom icon" message="This has a slotted icon" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <div class="alert-icon icon-neutral svg-container">
          <slot name="neutral-icon"></slot>
        </div>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              Using custom icon
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body">
              This has a slotted icon
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
  <svg fill="#000000" height="24px" slot="neutral-icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0V0z" fill="none"></path>
    <path d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"></path>
  </svg>
</ic-alert>
`;

exports[`ic-alert component should render default icon of success variant when the show-default-icon prop is set to false 1`] = `
<ic-alert class="ic-alert-no-default-icon ic-alert-variant-success" heading="Success" message="This alert has the default success icon" role="alert" show-default-icon="false" variant="success" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-success">
      <div class="divider divider-success"></div>
      <div class="alert-content">
        <span class="alert-icon icon-success svg-container">
          svg
        </span>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              Success
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body">
              This alert has the default success icon
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
</ic-alert>
`;

exports[`ic-alert component should render with a dismiss icon when the prop is applied 1`] = `
<ic-alert class="ic-alert-variant-neutral" dismissible="true" message="This is dismissible" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <slot name="message">
            <ic-typography variant="body">
              This is dismissible
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container">
        <ic-button class="dismiss-icon svg-container" theme="dark" title="Dismiss" variant="icon-tertiary">
          svg
        </ic-button>
      </div>
    </div>
  </template>
</ic-alert>
`;

exports[`ic-alert component should render with a heading when supplied 1`] = `
<ic-alert class="ic-alert-variant-neutral" heading="Test heading" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              Test heading
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body"></ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
</ic-alert>
`;

exports[`ic-alert component should render with a message when supplied 1`] = `
<ic-alert class="ic-alert-variant-neutral" message="Test message" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <slot name="message">
            <ic-typography variant="body">
              Test message
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
</ic-alert>
`;

exports[`ic-alert component should render with no icon when the show-default-icon prop is set to false 1`] = `
<ic-alert class="ic-alert-variant-neutral" heading="No icon" message="This alert has no icon" role="alert" show-defaul-icon="false" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              No icon
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body">
              This alert has no icon
            </ic-typography>
          </slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
</ic-alert>
`;

exports[`ic-alert component should test rendering an action after initial render 1`] = `
<ic-alert class="ic-alert-variant-neutral" heading="Test heading" role="alert" style="--ic-alert-min-height: 3.5rem;">
  <template shadowrootmode="open">
    <div class="container container-neutral">
      <div class="divider divider-neutral"></div>
      <div class="alert-content">
        <span class="alert-icon icon-neutral svg-container">
          svg
        </span>
        <div class="alert-message">
          <ic-typography class="alert-title" variant="subtitle-large">
            <p>
              Test heading
            </p>
          </ic-typography>
          <slot name="message">
            <ic-typography variant="body"></ic-typography>
          </slot>
        </div>
        <div class="alert-action-container">
          <slot name="action"></slot>
        </div>
      </div>
      <div class="dismiss-icon-container"></div>
    </div>
  </template>
  <button slot="action"></button>
</ic-alert>
`;
